api/src/main/java/org/apache/iceberg/encryption/NativeFileDecryptParameters.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Barebone decryption parameters, one object per content file.
+ * Carries the file and column encryption keys (and optional AAD prefix).
+ */
+public class NativeFileDecryptParameters {
+  private ByteBuffer fileAadPrefix;
+  private Map<String, ByteBuffer> columnKeys;
+  private ByteBuffer fileKey;
+
+  private NativeFileDecryptParameters(Map<String, ByteBuffer> columnKeys, ByteBuffer fileKey,
+                                      ByteBuffer fileAadPrefix) {
+    Preconditions.checkState((columnKeys != null && columnKeys.size() > 0) || fileKey != null,
+            "No file or column keys are supplied");
+    this.columnKeys = columnKeys;
+    this.fileKey = fileKey;
+    this.fileAadPrefix = fileAadPrefix;
+  }
+
+  /**
+   * Creates the builder.
+   * @param fileKey per-file encryption key. For example, used as "footer key" DEK in Parquet encryption.
+   */
+  public static Builder create(ByteBuffer fileKey) {
```

Comment:
Not clear why we need fileKey.

Reply:
in Parquet, we need it for setting the file footer key.